### PR TITLE
simplify & fix .create_mofa_from_seurat

### PR DESCRIPTION
**Pull request by jan-glx**
These are ~~three~~ two separate improvements, pick what you like.

### CODE REVIEW

File: MOFA2/R/create_mofa.R [hunk 1]
```diff
@@ -274,14 +274,12 @@ create_mofa <- function(data, groups = NULL, ...) {
...
+    if (!all(names(features) %in% assays)) {
         stop("Please make sure all the names of the features list correspond to views (assays) names being used for the model")
     }
   } else {
-    if (!is.null(features) && (length(assays) > 1))
-       stop("When using multiple assays, subset features with a list with corresponding views (assays) names")
-    # Make a list out of a vector
-    features <- list(features)
+    # replicate features for each assay
+    features <- rep(list(features), length(assays))
```
[the remark this answers is not in the document]
Comment (jan-glx) [reply MOFA2/R/create_mofa.R:282]: I have yet used MOFA in this way. I though MOFA would implement some functionality for when not only the samples but also features represent a shared dimension ( as with e.g. RNA & RNA Velocity). If this is not the case throwing an error in this case might indeed avoid confusion.

[the remark this answers is not in the document]
Comment (jan-glx) [reply MOFA2/R/create_mofa.R:282]: should I revert 72cbd8cd11a905ad7fca888fefbbbd68f2ef7ef9 ?

File: MOFA2/R/create_mofa.R [hunk 2]
```diff
@@ -320,16 +320,12 @@ create_mofa <- function(data, groups = NULL, ...) {
 
 # (Hidden) function to split a list of matrices into a nested list of matrices
 .split_data_into_groups <- function(data, groups) {
-  groups_names <- unique(groups)
-  tmp <- lapply(data, function(x) {
-    tmp_view <- lapply(groups_names, function(p) {
-      x[,groups == p]
+  group_indices <- split(seq_along(groups), factor(groups, exclude = character(0))) # factor call avoids dropping NA
```
[the remark this answers is not in the document]
Comment (gtca) [reply MOFA2/R/create_mofa.R:323]: Alright, I've added a basic test in 9d77dce. So we should be fine going ahead with it.

File: MOFA2/R/create_mofa.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (gtca) [reply MOFA2/R/create_mofa.R:282]: Due to not having features with the same name by design, probably yes, and we could merge other two commits.

**Comment by jan-glx** [comment]
I got rid of 72cbd8cSo, is this good to go now?

**Pull request merged**